File: Analisis_data_shopee/shopee/5.py
import pandas as pd
import mysql.connector
import numpy as np
import matplotlib.pyplot as plt

connection = mysql.connector.connect(host='localhost', user='root', password='', database='pembelian_shopee')
query1 = "SELECT*FROM order_detail"
query2 = "SELECT*FROM sku_detail"
df1 = pd.read_sql(query1, connection)
df2 = pd.read_sql(query2, connection)
connection.close()

df = pd.merge(df1, df2, left_on='sku_id', right_on='id', how='inner')
df = df[df['is_valid'] == 1]
# df['order_date'] = pd.to_datetime(df['order_date'])
# df = df[df['order_date'].dt.year == 2022]
print(df)

df['produk'] = np.select(
    [df['sku_name'].str.lower().str.contains('samsung'),
    df['sku_name'].str.lower().str.contains('apple')|df['sku_name'].str.lower().str.contains('iphone')|df['sku_name'].str.lower().str.contains('macbook'),
    df['sku_name'].str.lower().str.contains('sony'),
    df['sku_name'].str.lower().str.contains('huawei'),
    df['sku_name'].str.lower().str.contains('lenovo')
    ],
    ['Samsung', 'Apple', 'Sony', 'Huawei', 'Lenovo'],
    default='others'
)
df = df[df['produk'] != 'others']
print(df)
hasil = df.groupby('produk')['after_discount'].sum().round().reset_index(name='total').sort_values('total', ascending=False).reset_index(drop=True)
print(hasil)

File: Analisis_data_shopee/shopee/5.r
library(RMySQL)
library(dplyr)
library(tidyr)
library(lubridate)

# Ganti 'dbname', 'username', 'password', dan 'host' sesuai dengan informasi database Anda
con <- dbConnect(MySQL(), dbname = "pembelian_shopee", user = "root", password = "", host = "localhost")

query1 <- "SELECT * FROM order_detail"
query2 <- "SELECT * FROM sku_detail"

df1 <- dbGetQuery(con, query1)
df2 <- dbGetQuery(con, query2)
dbDisconnect(con)

print(head(df1))
cat("\n")
print(head(df2))
cat("\n\n")

# df <- merge(df1, df2, by.x = "sku_id", by.y = "id", all=FALSE)
df <- inner_join(df1, df2, by = c("sku_id" = "id"), relationship = "many-to-many")
print(head(df))
cat("\n\n")

df <- filter(df, is_valid ==1)
df <- df %>% mutate(
    produk = case_when(
        grepl('samsung', tolower(sku_name)) ~ 'samsung',
        grepl('apple|iphone|macbook', tolower(sku_name)) ~ 'apple',
        grepl('sony', tolower(sku_name)) ~ 'sony',
        grepl('huawei', tolower(sku_name)) ~ 'huawei',
        grepl('lenovo', tolower(sku_name)) ~ 'lenovo',
        TRUE ~ 'others'
    )
)

df <- filter(df, produk != 'others')
print(head(df))

hasil <- df %>% group_by(produk) %>% summarise(total = sum(after_discount)) %>% arrange(desc(total))
print(hasil)